spec/ruby/core/io/gets_spec.rb

```diff
@@ -113,6 +113,37 @@
         $..should == @count += 1
       end
     end
+
+    describe "that consists of multiple bytes" do
+      it "should match the separator even if the buffer is filled over subsequent reads" do
+        begin
+          read, write = IO.pipe
```

Comment:
Minor, but if `IO.pipe` would fail then we'd get a NoMethodError in the `ensure`, so it would be better to move those before the `begin`, or to use the block form of `IO.pipe` which automatically closes them at the end.

Reply:
Thanks. I didn't realize there was a block form. That cleans things up.